src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -833,6 +835,31 @@ def load_module(name, value):
 
         init_dict = {k: v for k, v in init_dict.items() if load_module(k, v)}
 
+        for key in init_dict.keys():
+            if key not in passed_class_obj:
+                continue
+
+            class_obj = passed_class_obj[key]
+            _expected_class_types = []
+            for expected_type in expected_types[key]:
+                if isinstance(expected_type, enum.EnumMeta):
+                    _expected_class_types.extend(expected_type.__members__.keys())
+                else:
+                    _expected_class_types.append(expected_type.__name__)
+
+            _is_valid_type = class_obj.__class__.__name__ in _expected_class_types
+            if isinstance(class_obj, SchedulerMixin) and not _is_valid_type:
+                _requires_flow_match = any("FlowMatch" in class_type for class_type in _expected_class_types)
```

Comment:
I think checking against a `FlowMatchSchedulers` enum would be better in case we end up not using "FlowMatch" in the class name. 

```python
_requires_flow_match = any(class_type in FlowMatchSchedulers.__members__ for class_type in _expected_class_types)
_is_flow_match =  class_obj.__class__.__name__ in FlowMatchSchedulers
```  

cc: @yiyixuxu